core/src/main/java/io/substrait/dsl/SubstraitBuilder.java

```diff
@@ -57,35 +57,63 @@ public SubstraitBuilder(SimpleExtension.ExtensionCollection extensions) {
   }
 
   // Relations
+  public Aggregate.Measure measure(AggregateFunctionInvocation aggFn) {
+    return Aggregate.Measure.builder().function(aggFn).build();
+  }
+
+  public Aggregate.Measure measure(AggregateFunctionInvocation aggFn, Expression preMeasureFilter) {
+    return Aggregate.Measure.builder().function(aggFn).preMeasureFilter(preMeasureFilter).build();
+  }
+
   public Aggregate aggregate(
       Function<Rel, Aggregate.Grouping> groupingFn,
-      Function<Rel, List<AggregateFunctionInvocation>> measuresFn,
+      Function<Rel, List<AggregateFunctionInvocation>> aggsFn,
+      Rel input) {
+    Function<Rel, List<Aggregate.Measure>> measuresFn =
+        aggsFn.andThen(
+            functionInvocations ->
+                functionInvocations.stream().map(this::measure).collect(Collectors.toList()));
+    return aggregateM(groupingFn, measuresFn, input);
+  }
+
+  public Aggregate aggregateM(
+      Function<Rel, Aggregate.Grouping> groupingFn,
+      Function<Rel, List<Aggregate.Measure>> measuresFn,
       Rel input) {
```

Comment:
I tried to add this as an overload of `aggregate` like:
```java
  public Aggregate aggregate(
      Function<Rel, Aggregate.Grouping> groupingFn,
      Function<Rel, List<Aggregate.Measure>> measuresFn,
      Rel input) {
```
but it results in clashes as both methods have the same erasure.